lib/eclipse/EclipseState/Schedule/TimeMap.cpp

```diff
@@ -88,36 +85,34 @@ namespace Opm {
     {
         if (m_timeList.size() < 2)
             return 0.0;
-        boost::posix_time::time_duration deltaT = m_timeList.back() - m_timeList.front();
-        return static_cast<double>(deltaT.total_milliseconds())/1000.0;
+        const time_t deltaT = m_timeList.back() - m_timeList.front();
+        return static_cast<double>(deltaT);
     }
 
-
-    void TimeMap::addTime(boost::posix_time::ptime newTime) {
-        boost::posix_time::ptime lastTime = m_timeList.back();
-        size_t step = m_timeList.size();
+    void TimeMap::addTime(std::time_t newTime) {
+        const std::time_t lastTime = m_timeList.back();
+        const size_t step = m_timeList.size();
         if (newTime > lastTime) {
-            boost::gregorian::date new_date = newTime.date();
-            boost::gregorian::date prev_date = lastTime.date();
+            int new_day, new_month, new_year, last_day, last_month, last_year;
+            util_set_date_values_utc(newTime, &new_day, &new_month, &new_year);
+            util_set_date_values_utc(lastTime, &last_day, &last_month, &last_year);
 
-            if (new_date.month() != prev_date.month())
-              m_first_timestep_months.push_back(step);
+            if (new_month != last_month)
+                m_first_timestep_months.push_back(step);
 
-            if (new_date.year() != prev_date.year())
-                m_first_timestep_years.push_back( step );
+            if (new_year != last_year)
+                m_first_timestep_years.push_back(step);
 
-            m_timeList.push_back( newTime );
+            m_timeList.push_back(newTime);
         } else
             throw std::invalid_argument("Times added must be in strictly increasing order.");
     }
 
-
-    void TimeMap::addTStep(boost::posix_time::time_duration step) {
-        boost::posix_time::ptime newTime = m_timeList.back() + step;
+    void TimeMap::addTStep(int64_t step) {
+        const std::time_t newTime = m_timeList.back() + step;
```

Comment:
Instead of explicitly using `+` I would recommend the `TimeMap::forward( )` method:
```
addTime( forward( m_timeList.back() , step ));
```
that way we are - at least slightly - better prepared to introduce a non-scalar time type.